src/SmartTransactionsController.ts

```diff
@@ -322,23 +322,30 @@ export default class SmartTransactionsController extends StaticIntervalPollingCo
   }
 
   async _executePoll({
-    networkClientId,
+    chainIds,
   }: SmartTransactionsControllerPollingInput): Promise<void> {
     // if this is going to be truly UI driven polling we shouldn't really reach here
     // with a networkClientId that is not supported, but for now I'll add a check in case
     // wondering if we should add some kind of predicate to the polling controller to check whether
     // we should poll or not
-    const chainId = this.#getChainId({ networkClientId });
-    if (!this.#supportedChainIds.includes(chainId)) {
+    // const chainId = this.#getChainId({ networkClientId });
+
+    const filteredChainIds = chainIds.filter((chainId) =>
```

Comment:
Will `chainIds` always be an array? If it could be undefined, this would throw an error `Cannot read properties of undefined (reading 'filter')`